File: examples/companion-packages/dotnet-csharp-app/src/AgentXM.Examples.PawMatch.CSharp/Flags.cs
using AgentXM.Examples.TinyFlags.CSharp;
using TinyFlagsClient = AgentXM.Examples.TinyFlags.CSharp.TinyFlags;

namespace AgentXM.Examples.PawMatch.CSharp;

internal static class PawMatchFlags
{
    public const string HomeCheckFollowup = "home-check-followup";
    public const string FeeBreakdownDetailed = "fee-breakdown-detailed";
    public const string LongStayHighlight = "long-stay-highlight";
    public const string SuggestDonateAfterAdoption = "suggest-donate-after-adoption";
    public const string ShowCharityRatings = "show-charity-ratings";
    public const string RecommendationStrategy = "recommendation-strategy";
    public const string MatchQuizDepth = "match-quiz-depth";
    public const string PetCardStyle = "pet-card-style";
    public const string DonateFocusDefault = "donate-focus-default";

    public static TinyFlagsClient Create() => TinyFlagsClient.Create(new Dictionary<string, FlagDefinition>
    {
        [HomeCheckFollowup] = TinyFlag.Boolean(defaultValue: false, rollout: 25),
        [FeeBreakdownDetailed] = TinyFlag.Boolean(defaultValue: true),
        [LongStayHighlight] = TinyFlag.Boolean(defaultValue: true),
        [SuggestDonateAfterAdoption] = TinyFlag.Boolean(defaultValue: false, rollout: 50),
        [ShowCharityRatings] = TinyFlag.Boolean(defaultValue: true),
        [RecommendationStrategy] = TinyFlag.Variant(
            ["popularity", "match-quiz", "longest-stay"],
            defaultValue: "match-quiz",
            rollout: new Dictionary<string, int> { ["longest-stay"] = 20 }),
        [MatchQuizDepth] = TinyFlag.Variant(
            ["short", "standard", "thorough"],
            defaultValue: "standard"),
        [PetCardStyle] = TinyFlag.Variant(
            ["compact", "detailed", "playful"],
            defaultValue: "detailed"),
        [DonateFocusDefault] = TinyFlag.Variant(
            ["all", "shelters", "rescue"],
            defaultValue: "all"),
    });
}

File: examples/companion-packages/dotnet-fsharp-app/src/AgentXM.Examples.PawMatch.FSharp/Flags.fs
module AgentXM.Examples.PawMatch.FSharp.Flags

open AgentXM.Examples.TinyFlags.FSharp

[<Literal>]
let HomeCheckFollowup = "home-check-followup"

[<Literal>]
let FeeBreakdownDetailed = "fee-breakdown-detailed"

[<Literal>]
let LongStayHighlight = "long-stay-highlight"

[<Literal>]
let SuggestDonateAfterAdoption = "suggest-donate-after-adoption"

[<Literal>]
let ShowCharityRatings = "show-charity-ratings"

[<Literal>]
let RecommendationStrategy = "recommendation-strategy"

[<Literal>]
let MatchQuizDepth = "match-quiz-depth"

[<Literal>]
let PetCardStyle = "pet-card-style"

[<Literal>]
let DonateFocusDefault = "donate-focus-default"

let create () =
    TinyFlags.create [
        HomeCheckFollowup, Flag.Boolean(defaultValue = false, rollout = 25)
        FeeBreakdownDetailed, Flag.Boolean(defaultValue = true)
        LongStayHighlight, Flag.Boolean(defaultValue = true)
        SuggestDonateAfterAdoption, Flag.Boolean(defaultValue = false, rollout = 50)
        ShowCharityRatings, Flag.Boolean(defaultValue = true)
        RecommendationStrategy,
        Flag.Variant(
            variants = [ "popularity"; "match-quiz"; "longest-stay" ],
            defaultValue = "match-quiz",
            rollout = Map.ofList [ "longest-stay", 20 ]
        )
        MatchQuizDepth,
        Flag.Variant(variants = [ "short"; "standard"; "thorough" ], defaultValue = "standard")
        PetCardStyle,
        Flag.Variant(variants = [ "compact"; "detailed"; "playful" ], defaultValue = "detailed")
        DonateFocusDefault,
        Flag.Variant(variants = [ "all"; "shelters"; "rescue" ], defaultValue = "all")
    ]
